assistant/assistant.py

```diff
@@ -16,40 +16,35 @@
 MSG_RESPONSE = "response"
 MSG_STATUS = "status"
 MSG_SUCCESS = "success"
-OLLAMA_HOST = os.getenv("OLLAMA_HOST")
-OLLAMA_PORT = os.getenv("OLLAMA_PORT")
 
-# --- Ollama section
+# --- LLM section
 
 
 class LLM:
     def __init__(self, chat_model: str):
-        self.llm_client = Client(host=f"http://{OLLAMA_HOST}:{OLLAMA_PORT}")
         self.chat_model = chat_model
-        pull_response = self.llm_client.pull(self.chat_model)
-        if pull_response[MSG_STATUS] != MSG_SUCCESS:
-            raise Exception(
-                f"failed to pull {self.chat_model}: {pull_response}"
-            )
-
+        self.client = OpenAI(
+            # This is the default and can be omitted
+            api_key=os.environ.get("OPENAI_API_KEY"),
+        )
         self.messages = []
         self.append_message(ROLE_SYSTEM, os.getenv("ASSISTANT_ROLE"))
 
     def chat(self, content: str) -> dict:
         self.append_message(ROLE_USER, content)
-        response = self.llm_client.chat(
+        response = self.client.chat.completions.create(
             model=self.chat_model, messages=self.messages
         )
-        message = response[MSG_MESSAGE]
-        self.append_message(message[MSG_ROLE], message[MSG_CONTENT])
-        return response
+        logging.info(f"OPENAI RESPONSE: {response.choices[0].message.content}")
+        message = response.choices[0].message
+        self.append_message(message.role, message.content)
+        return message.content
 
     def append_message(self, role, content):
-        self.messages.append({MSG_ROLE: role, MSG_CONTENT: content})
-        # logging.debug(f"------- {self.messages}")
+        self.messages.append({"role": role, "content": str(content)})
```

Comment:
If you don't want to use the constants I defined for that I am fine with that (I was thinking to remove them anyway), but please make the code consistent. Either do or do not use the constants and do the same everywhere. I think using magic strings instead of constants is more common in the Python ecosystem. 🤷